identity-apps-core/components/org.wso2.identity.apps.common/src/test/java/org/wso2/identity/apps/common/listener/AppPortalRoleManagementListenerTest.java

```diff
@@ -0,0 +1,201 @@
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.ORGANIZATION;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.RoleTableColumns.ROLE_NAME;
+import static org.wso2.identity.apps.common.util.AppPortalConstants.CONSOLE_APP;
+
+/**
+ * Test class for AppPortalRoleManagementListener test cases.
+ */
+public class AppPortalRoleManagementListenerTest {
+
+    private AutoCloseable closeable;
+
+    private MockedStatic<PrivilegedCarbonContext> privilegedCarbonContext;
+    private MockedStatic<OrganizationManagementUtil> organizationManagementUtil;
+    private MockedStatic<AppsCommonDataHolder> appsCommonDataHolder;
+
+    @Mock
+    private UserRealm mockUserRealm;
+
+    @Mock
+    private RoleManagementService roleManagementService;
+
+    @Mock
+    private Role role;
+
+    @BeforeMethod
+    public void setUp() {
+
+        setUpCarbonHome();
+        privilegedCarbonContext = mockStatic(PrivilegedCarbonContext.class);
+        organizationManagementUtil = mockStatic(OrganizationManagementUtil.class);
+        appsCommonDataHolder = mockStatic(AppsCommonDataHolder.class);
+        mockCarbonContext(privilegedCarbonContext);
+        closeable = MockitoAnnotations.openMocks(this);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        privilegedCarbonContext.close();
+        organizationManagementUtil.close();
+        appsCommonDataHolder.close();
+        closeable.close();
+    }
+
+    @DataProvider(name = "preUpdateUserListOfRoleDataProvider")
+    public Object[][] preUpdateUserListOfRoleDataProvider() {
+        return new Object[][]{
+            // Test case where deletedUserIDList == null
+            {"roleId", Collections.emptyList(), null, CARBON_HOME, true, false, "adminUserId"},
+
+            // Test case where !isAdministratorRole
+            {"roleId", Collections.emptyList(), Collections.singletonList("deletedUser"), CARBON_HOME, false, false,
+                "adminUserId"},
+
+            // Test case where isOrganization == true
+            {"roleId", Collections.emptyList(), Collections.singletonList("deletedUser"), CARBON_HOME, true, true,
+                "adminUserId"},
+
+            // Test case where deletedUserIDList contains adminUserId
+            {"roleId", Collections.emptyList(), Collections.singletonList("adminUserId"), CARBON_HOME, true, false,
+                "adminUserId"}
+        };
+    }
+
+    @Test(dataProvider = "preUpdateUserListOfRoleDataProvider")
+    public void testPreUpdateUserListOfRole(String roleId, List<String> newUserIDList, List<String> deletedUserIDList,
+                                String tenantDomain, boolean isAdminRole, boolean isOrganization, String adminUserId)
+        throws Exception {
+
+        AppPortalRoleManagementListener appPortalRoleManagementListener = spy(
+            new AppPortalRoleManagementListener(true));
+
+        // Set up the mock behaviors
```

Comment:
Add a full stop at the end of comment lines.Fix other places as well

Reply:
Added full stops to all comments.